src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -76,6 +76,32 @@ class TezosNodeOperator(val node: TezosRPCInterface, val network: String, batchC
   type BallotBlock = (Block, List[Voting.Ballot])
   type BakerBlock = (Block, List[Voting.BakerRolls])
 
+  /**
+    * Generic fetch for paginated data relative to a specific block.
+    * Will return back the entities indexed by their unique key
+    * from those passed as inputs, through the use of the
+    * `entityLoad` function, customized per each entity type.
+    * e.g. we use it to load accounts and delegates without
+    * duplicating any logic.
+    */
+  def getPaginatedEntitiesForBlock[Key, Entity](
+    entityLoad: (List[Key], BlockHash) => Future[Map[Key, Entity]]
```

Comment:
Rename to 'getEntityForBlock', since that's the shape of the name used for this function in general.

Reply:
I'm not sure I follow, this method concern is specifically to load lazily evaluated "pages" of some type, that's the added behaviour over the passed-in loader `entityLoad`.
And that the reason for the unusual return type: `Iterator[Future[...]]` which loads async pages of results on demand